diff --git a/arch/x86/kernel/entry.asm b/arch/x86/kernel/entry.asm
index 5aa1966..04f3c43 100644
--- a/arch/x86/kernel/entry.asm
+++ b/arch/x86/kernel/entry.asm
@@ -114,15 +114,6 @@ boot_pgt:
 SECTION .ktext
 align 4
 start64:
-    ; [for test only]: create pseudo hbmem
-    mov rcx, QWORD [limit]
-    sub rcx, 0x10000000
-    mov QWORD [limit], rcx
-    mov [hbmem_base], rcx
-    mov rcx, 0x10000000
-    mov [hbmem_size], rcx
-
-
     ; reset registers to kill any stale realmode selectors
     xor eax, eax
     mov ds, eax
