diff --git a/kernel/syscall.c b/kernel/syscall.c
index 13fcc42..0358ecf 100644
--- a/kernel/syscall.c
+++ b/kernel/syscall.c
@@ -300,6 +300,9 @@ ssize_t sys_sbrk(ssize_t incr)
 		}
 	} else ret = -ENOMEM;
 
+	LOG_INFO("sbrk: old = %#lx, new = %#lx\n", PAGE_CEIL(ret), PAGE_FLOOR(heap->end) );
+    vma_dump();
+
 	// allocation and mapping of new pages for the heap
 	// is catched by the pagefault handler
 
@@ -335,6 +338,9 @@ ssize_t sys_hbmem_sbrk(ssize_t incr)
 		}
 	} else ret = -ENOMEM;
 
+	LOG_INFO("sbrk: old = %#lx, new = %#lx\n", PAGE_CEIL(ret), PAGE_FLOOR(heap->end) );
+    vma_dump();
+
 	// allocation and mapping of new pages for the heap
 	// is catched by the pagefault handler
 
